### ISSUE TITLE
Source Remapping

### ISSUE THREAD

**Issue by CoolMineman** [CoolCrabs/brachyura#1]
Source remapping would be useful for publishing sources jars and consuming sources jars from loom. None of the options seem pleasant
Some options:

eclipse jdt based - dropped java 8 support and I'd like to keep targeting java 8, also doesn't use nio but that's not the end of the world. existing tools exist.

javaparser based - lexical preserving parsing seems to be buggy see javaparse/javaparser#3284 additionally the api is fairly clunky. I pushed some progress on a javaparser tool to a separate branch but I don't want to continue it rn

simple find and replace - can only consume sources jars. easyish to implement.

semanicdb - needs further evaluation https://github.com/scalameta/scalameta/blob/main/semanticdb/semanticdb3/guide.md#javac-compiler-plugin

Other (bad) options:
own javac plugin
basically own java compiler front end
